Content.Server/Sticky/Systems/StickySystem.cs

```diff
@@ -18,19 +19,48 @@ public sealed class StickySystem : EntitySystem
     [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
     [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
 
+
     private const string StickerSlotId = "stickers_container";
 
     public override void Initialize()
     {
         base.Initialize();
         SubscribeLocalEvent<StickSuccessfulEvent>(OnStickSuccessful);
         SubscribeLocalEvent<UnstickSuccessfulEvent>(OnUnstickSuccessful);
+
         SubscribeLocalEvent<StickyComponent, AfterInteractEvent>(OnAfterInteract);
         SubscribeLocalEvent<StickyComponent, GetVerbsEvent<Verb>>(AddUnstickVerb);
+
+        SubscribeLocalEvent<StickyComponent, StickyComponentTestAttemptEvent>(OnStickySystemTest);
+        SubscribeLocalEvent<HasEntityStuckOnComponent, HasEntityStuckOnComponentTestAttemptEvent>(OnHasEntityStuckOnComponentTest);
+
+    }
+
+    private void OnHasEntityStuckOnComponentTest(EntityUid uid, HasEntityStuckOnComponent component, HasEntityStuckOnComponentTestAttemptEvent args)
+    {
+        if (EntityManager.HasComponent<HasEntityStuckOnComponent>(uid))
+        {
+            _popupSystem.PopupEntity(Loc.GetString(args.FailPopupMessage), uid ,Filter.Entities(args.User));
+            args.Cancel();
+        }
+    }
+
+    private void OnStickySystemTest(EntityUid uid, StickyComponent component, StickyComponentTestAttemptEvent args)
+    {
+
+        if (EntityManager.TryGetComponent(uid, out StickyComponent stickyComponent)
+            && stickyComponent.StuckTo != null)
+        {
+            _popupSystem.PopupEntity(Loc.GetString(args.FailPopupMessage), uid ,Filter.Entities(args.User));
+            args.Cancel();
+        }
     }
```

Comment:
As above, this is pretty much the opposite of what you want to do for the eventbus. You're raising events that just get used by 1 system instead of calling 1 system that raises events.

Reply:
I am not really sure how to achieve that ? Can you give me an example ?

Reply:
How you had it before [2168526](https://github.com/space-wizards/space-station-14/pull/7701/commits/2168526f5a4f8267314d27d179a066f597698d98) was better. Just needed to have the events handled in the right systems.